core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -321,6 +323,20 @@ public ApplyFeatureSetResponse applyFeatureSet(FeatureSetProto.FeatureSet newFea
               .build();
     }
 
+    for (Store store : storeRepository.findAll()) {
+      List<Subscription> subscriptionList = store.getSubscriptions();
+      String projectName = newFeatureSet.getSpec().getProject();
+      String featureSetName = newFeatureSet.getSpec().getName();
+      boolean isSubscribed =
+          isSubscribedToFeatureSet(subscriptionList, projectName, featureSetName);
+
+      if (!isSubscribed) {
+        throw new RegistrationException(
```

Comment:
I don't understand, why featureSet blacklisted by one storage cannot be used by another?
Here we allow only feature sets subscripted by all stores, which is not true already: we have sauron features sets, and shared redis is not subscribed to them.